drivers/pwm/pwm_stm32.c

```diff
@@ -656,27 +676,36 @@ static void pwm_stm32_irq_config_func_##index(const struct device *dev)        \
 replaced by 'st,prescaler' property in parent node, aka timers"
 #endif
 
-#define PWM_DEVICE_INIT(index)                                                 \
-	static struct pwm_stm32_data pwm_stm32_data_##index;                   \
-	IRQ_CONFIG_FUNC(index)						       \
-									       \
-	PINCTRL_DT_INST_DEFINE(index);					       \
-									       \
-	static const struct pwm_stm32_config pwm_stm32_config_##index = {      \
-		.timer = (TIM_TypeDef *)DT_REG_ADDR(DT_INST_PARENT(index)),    \
-		/* For compatibility reason, use pwm st_prescaler property  */ \
-		/* if exist, otherwise use parent (timers) property         */ \
-		.prescaler = DT_INST_PROP_OR(index, st_prescaler,              \
-			(DT_PROP(DT_INST_PARENT(index), st_prescaler))),       \
-		.pclken = DT_INST_CLK(index, timer),                           \
-		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(index),		       \
-		CAPTURE_INIT(index)					       \
-	};                                                                     \
-									       \
-	DEVICE_DT_INST_DEFINE(index, &pwm_stm32_init, NULL,                    \
-			    &pwm_stm32_data_##index,                           \
-			    &pwm_stm32_config_##index, POST_KERNEL,            \
-			    CONFIG_KERNEL_INIT_PRIORITY_DEVICE,                \
+#define DT_TIMER_COUNTER_MODE(inst)                                                  \
```

Comment:
Optional: Since this property is common to stm32 timer subnodes (counter_timer could use it as well), This could also be provided in an `include/dt-bindings/timer/timer_stm32.h` (name to be confirmed as this is not a real zephyr timer driver)

Reply:
Switched to `int` constants that are defined in `include/dt-bindings/timer/stm32-timer.h`. Filename is composed based on the names of the other `dt-bindings` files (e.g. `pinctrl`).